recaf-plugin/src/main/java/dev/xdark/recaf/plugin/RecafRootPlugin.java

```diff
@@ -0,0 +1,163 @@
+package dev.xdark.recaf.plugin;
+
+import dev.xdark.recaf.plugin.java.ZipPluginLoader;
+import me.coley.recaf.io.ByteSources;
+import me.coley.recaf.util.Directories;
+import me.coley.recaf.util.logging.Logging;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * Root plugin for Recaf, providing access to a {@link PluginManager} that reads from
+ * {@link Directories#getPluginDirectory()}.
+ *
+ * @author xtherk
+ */
+public class RecafRootPlugin {
```

Comment:
Still not 100% on this name, will probably rename again later once I start implementing UI integrations more.